tests/core/consensus/test_clique_consensus.py

```diff
@@ -183,48 +184,45 @@ def validate_seal_and_get_snapshot(clique, header):
 
 
 @pytest.fixture
-def clique(paragon_chain_with_clique):
-    _, clique = paragon_chain_with_clique
-    return clique
-
-
-@pytest.fixture
-def paragon_chain(paragon_chain_with_clique):
-    chain, _ = paragon_chain_with_clique
-    return chain
-
-
-@pytest.fixture
-def paragon_chain_with_clique(base_db):
+def paragon_chain(base_db):
 
     vms = ((0, PetersburgVM,),)
-
-    clique = CliqueConsensus(base_db)
-
-    vms = clique.amend_vm_configuration(vms)
+    clique_vms = CliqueApplier.amend_vm_configuration(vms)
 
     chain = MiningChain.configure(
-        vm_configuration=vms,
+        vm_configuration=clique_vms,
```

Comment:
So, I guess we would then also have a `NoConsensusApplier` and `PowConsensusApplier` and the canonical way to activate a chain for a specific type of consensus would be:

```python
Chain.configure(SomeApplier.amend_vm_configuration(vms))
```

But as I noted in a comment above, this setup seems to be setup to have each VM maintain their own `ConsensusAPI` instance which only seems to work if the `ConsensusAPI` were stateless.

Reply:
> So, I guess we would then also have a `NoConsensusApplier` and `PowConsensusApplier` and the canonical way to activate a chain for a specific type of consensus would be:

We don't seem to need it, because it's so trivial to just set the consensus class for the others, but I'm not really opposed. I think we can have a `SimpleConsensusApplier` that just changes the `consensus_class` and is sufficient for both PoW and NoProof.